File: src/main/java/com/dh/ondot/member/core/exception/NotFoundAnswerException.java
package com.dh.ondot.member.core.exception;

import com.dh.ondot.core.exception.NotFoundException;

import static com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_ANSWER;

public class NotFoundAnswerException extends NotFoundException {
    public NotFoundAnswerException(Long answerId) {
        super(NOT_FOUND_ANSWER.getMessage().formatted(answerId));
    }

    @Override
    public String getErrorCode() {
        return NOT_FOUND_ANSWER.name();
    }
}

File: src/main/kotlin/com/dh/ondot/member/core/exception/NotFoundAnswerException.kt
package com.dh.ondot.member.core.exception

import com.dh.ondot.core.exception.ErrorCode.NOT_FOUND_ANSWER
import com.dh.ondot.core.exception.NotFoundException

class NotFoundAnswerException(answerId: Long) :
    NotFoundException(NOT_FOUND_ANSWER.message.format(answerId)) {
    override val errorCode: String get() = NOT_FOUND_ANSWER.name
}
